web/src/composebox_typeahead.ts

```diff
@@ -960,20 +922,65 @@ export function get_candidates(
                 }
 
                 const stream_id = stream_data.get_stream_id(stream_name);
-                const topic_list = topics_seen_for(stream_id);
+                // We always show topic suggestions after the user types a stream,
+                // and let them pick between just showing the stream (the first option, when nothing follows ">")
+                // or adding a topic.
+                const topic_list = current_token.endsWith(">")
+                    ? [stream_name, ...topics_seen_for(stream_id)]
+                    : topics_seen_for(stream_id);
+
                 if (should_show_custom_query(token, topic_list)) {
                     topic_list.push(token);
                 }
                 const matcher = get_topic_matcher(token);
                 const matches = topic_list.filter((item) => matcher(item));
-                const matches_list: TopicSuggestion[] = matches.map((topic) => ({
-                    topic,
-                    type: "topic_list",
-                }));
+                const matches_list: TopicSuggestion[] = matches.map((topic) => {
+                    const stream = stream_data.get_sub_by_name(stream_name);
+                    assert(stream !== undefined);
+                    const is_stream_only = topic === stream_name;
```

Comment:
I would prefer to just have the stream element live in a different variable; you can just call the matcher function on a 1-element list of it directly, right?

Reply:
I think you should be able to just prepend the channel item onto `matches_list` at the end of this block (outside the loop) rather than doing all this hackery to have the channel name weirdly in the list of topics in the channel.

It can be a suggestion without being in that list the whole time.